src/EditorFeatures/Core/Implementation/Workspaces/ProjectCacheServiceFactory.cs

```diff
@@ -94,6 +108,28 @@ public IDisposable EnableCaching(ProjectId key)
                 }
             }
 
+            private bool PartOfP2PReferences(ProjectId key)
+            {
+                if (_activeCaches.Count == 0 || _workspace == null)
+                {
+                    return false;
+                }
+
+                var solution = _workspace.CurrentSolution;
+
+                foreach (var projectId in _activeCaches.Keys)
+                {
+                    // this should be cheap. graph is cached everytime project reference is updated.
+                    var p2pReferences = (ImmutableHashSet<ProjectId>)solution.GetProjectDependencyGraph().GetProjectsThatThisProjectTransitivelyDependsOn(projectId);
```

Comment:
Good catch

Sent from my Windows Phone
________________________________
From: Paul Harrington<mailto:notifications@github.com>
Sent: ‎6/‎4/‎2015 6:04 PM
To: dotnet/roslyn<mailto:roslyn@noreply.github.com>
Cc: HeeJae Chang<mailto:Heejae.Chang@microsoft.com>
Subject: Re: [roslyn] don't put p2p reference to implicit cache. (#3319)


In src/EditorFeatures/Core/Implementation/Workspaces/ProjectCacheServiceFactory.cs<https://github.com/dotnet/roslyn/pull/3319#discussion_r31781005>:

> @@ -94,6 +108,28 @@ public IDisposable EnableCaching(ProjectId key)
>                  }
>              }
>
> +            private bool PartOfP2PReferences(ProjectId key)
> +            {
> +                if (_activeCaches.Count == 0 || _workspace == null)
> +                {
> +                    return false;
> +                }
> +
> +                var solution = _workspace.CurrentSolution;
> +
> +                foreach (var projectId in _activeCaches.Keys)
> +                {
> +                    // this should be cheap. graph is cached everytime project reference is updated.
> +                    var p2pReferences = (ImmutableHashSet<ProjectId>)solution.GetProjectDependencyGraph().GetProjectsThatThisProjectTransitivelyDependsOn(projectId);


Does it make sense to hoist solution.GetProjectDependencyGraph() out of the loop?

—
Reply to this email directly or view it on GitHub<https://github.com/dotnet/roslyn/pull/3319/files#r31781005>.